p2panda-core/src/serde.rs

```diff
@@ -258,9 +258,15 @@ where
                     .map_err(|_| SerdeError::custom("invalid previous links, expected array"))?
                     .ok_or(SerdeError::custom("previous array missing"))?;
 
-                let extensions: Option<E> = seq
-                    .next_element()
-                    .map_err(|err| SerdeError::custom(format!("invalid extensions: {err}")))?;
+                let extensions: E = if std::mem::size_of::<E>() > 0 {
+                    seq.next_element()
+                        .map_err(|err| SerdeError::custom(format!("invalid extensions: {err}")))?
+                        .ok_or(SerdeError::custom("extensions missing"))?
+                } else {
+                    // For zero-sized types, we don't need to read anything from the sequence
+                    // as the serializer doesn't write anything for them
+                    unsafe { std::mem::zeroed() }
```

Comment:
Maybe there is a "safer" way to get a zero-sized value here but I haven't found it yet.

Reply:
Skipping the field is no problem, I think the missing piece is a safe way to populate `E` without knowing what it is. The riddle is, what do `()`, `struct S`, `enum E { E(()) }`, etc. all have in common?

Reply:
I can look into it a bit, maybe what is there now really is safe. The type is guaranteed to be zero bytes in this case after all.

Reply:
looking into ZSTs some more (zero-sized types) it looks like this is the way. There is this https://github.com/rust-lang/rust/pull/146479 to look forward to but I guess it won't be stable for a while.

My takeaway at doing this unsafely is that it's fine as long as the ZST is not uninhabited, like `enum E {}`, but in that case you couldn't serialize it in the first place.

What do you think, is it acceptable to just have this extra byte hanging around for now? Will anyone be using a zero sized extension type in the real world anyway?